lib/find-author-name.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const parse = require('parse-github-url');
+
+module.exports = ({repoURL, dirname}) => {
+	if (repoURL) {
+		return parse(repoURL).owner;
+	}
+
+	const pkgJsonFile = path.resolve(dirname, 'package.json');
+	if (fs.existsSync(pkgJsonFile)) {
+		const json = require(pkgJsonFile);
+		if (json && json.repository) {
```

Comment:
You are making assumption about the format of `repository` that are not always true. `repository` can have many different formats.

You're also making assumption that the package.json is in the same directory as the readme.

Use https://github.com/sindresorhus/read-pkg-up to have the data normalized.

Reply:
Good suggestion. But I chose `read-pkg` instead of `read-pkg-up`.

This project has made the assumption that readme file is in the root directory of the project. So don't searching package.json in parent directories, and that will make writing test hard. Because `read-pkg-up` may find the package.json of awesome-lint itself and result in some misleadings.